app/javascript/components/async-credentials/async-credentials.jsx

```diff
@@ -23,98 +21,85 @@ const AsyncCredentials = ({
+    const state = formOptions.getFieldState(dependency);
+    return state && state.valid;
+  });
+
+  const onClick = () => {
     const { values } = formOptions.getState();
-    const currentValues = asyncFields.reduce((acc, curr) => {
-      set(acc, curr, get(values, curr));
-      return { ...acc };
-    }, {});
-    const valid = (isEqual(lastValid, currentValues) || isEqual(initialValues, currentValues)) ? undefined : false;
-    setAsyncError(validateDefaultError);
-    change(validateName, valid);
+    setState(state => ({ ...state, validating: true, errorMessage: undefined }));
+
+    asyncValidate(values, dependencies).then(() => {
+      formOptions.change(name, true);
+      setState(state => ({
+        ...state,
+        validating:
+        false,
+        lastValid: snapshot(values),
+        errorMessage: undefined,
+      }));
+    }).catch((error) => {
+      formOptions.change(name, undefined);
+      setState(state => ({ ...state, validating: false, errorMessage: error }));
+    });
   };
 
   return (
-    <Fragment>
+    <>
       {formOptions.renderForm(fields.map(field => ({
         ...field,
         ...(field.component === 'password-field' ? { parent: name, edit } : undefined),
-        isDisabled: field.isDisabled || validating,
-        onChange: value => enhancedChange(value, field.name, name, formOptions.change),
+        isDisabled: field.isDisabled,
       })), formOptions)}
-      <FieldProvider initialValue={!!edit} name={name} validate={value => (value === false ? asyncError : undefined)}>
-        {({ input, meta }) => (
-          <FormGroup validationState={meta.error ? 'error' : null}>
-            <input type="hidden" {...input} />
-            <CheckErrors
-              subscription={{ valid: true, invalid: true, active: true }}
-              names={[...asyncFields, ...validationDependencies]}
-              FieldProvider={FieldProvider}
-            >
-              {valid => (
-                <Fragment>
-                  <Button
-                    bsSize="small"
-                    bsStyle="primary"
-                    onClick={() => handleAsyncValidation(formOptions, name, [...asyncFields, ...validationDependencies])}
-                    disabled={valid.includes(false) || validating}
-                  >
-                    {validating ? validationProgressLabel : validateLabel}
-                    {validating && <ButtonSpinner /> }
-                  </Button>
-                  {!meta.error && !valid.includes(false) && !isEqual(lastValid, {}) && <HelpBlock>{ validationSuccessLabel }</HelpBlock>}
-                  {meta.error && <HelpBlock>{asyncError}</HelpBlock>}
-                </Fragment>
-              )}
-            </CheckErrors>
-          </FormGroup>
-        )}
-      </FieldProvider>
-    </Fragment>
+      <FormGroup validationState={meta.error ? 'error' : null}>
+        <input type="hidden" input={input} />
```

Comment:
`input={input}` => `{...input}` ?

Reply:
Well, we don't actually need it, but it's ugly, fixing...

Reply:
This might be a breaking change, can you please retest if the credential validation sets the valid state of the form properly?

Reply:
https://github.com/ManageIQ/manageiq-ui-classic/pull/7279/files#diff-4deb71aa6a0b588cdfcb0f11939f8acbL82

It was the same way originally, so shouldn't break it, but I'll test..

Reply:
Better safe than sorry :laughing: